kv/store.go

```diff
@@ -32,6 +32,37 @@ func NewStore(client etcd.KV, jobsBucketSize uint, log *zap.SugaredLogger) *Stor
 	}
 }
 
+// CreateCheckpoint initalizes a new checkpoint for a specific workerID
+func (store Store) CreateCheckpoint(key string) error {
+	_, err := store.client.Txn(context.TODO()).Then(etcd.OpPut(key+"/", "")).Commit()
+	return err
+}
+
+// RetrieveCheckpoint returns the existing checkpoint for a given workerID, or an error if not found
+func (store Store) RetrieveCheckpoint(key string) (string, error) {
+	checkpoint, err := store.client.Get(context.TODO(), key+"/", etcd.WithPrefix())
+	if checkpoint.Count == 0 {
+		return "", ErrKeyNotFound
+	}
+	if err != nil {
+		return "", err
+	}
+
+	return string(checkpoint.Kvs[0].Value), nil
+}
+
+// UpdateCheckpoint updates the current checkpoint information for a given workerID
+func (store Store) UpdateCheckpoint(key, value string) error {
+	_, err := store.client.Put(context.TODO(), key+"/", value)
+	return err
+}
+
+// DeleteCheckpoint removes the checkpoint altogether from the store when a job/worker is removed completely
+func (store Store) DeleteCheckpoint(key string) error {
```

Comment:
I'm not sure if we need this method. We can just delete checkpoint when we delete a connection, in a transaction. There is no situation that we need to delete connection but leave checkpoints.